lib/spack/spack/spec.py

```diff
@@ -1919,9 +1919,13 @@ def from_dict(data):
 
             yaml_deps = node[name]['dependencies']
             for dname, dhash, dtypes in Spec.read_yaml_dep_specs(yaml_deps):
+                dspec = DependencySpec(deps[name], deps[dname], dtypes)
+
                 # Fill in dependencies by looking them up by name in deps dict
-                deps[name]._dependencies[dname] = DependencySpec(
-                    deps[name], deps[dname], dtypes)
+                deps[name]._dependencies[dname] = dspec
+
+                # Fill in dependents by looking them up by dname in deps dict
+                deps[dname]._dependents[name] = dspec
```

Comment:
Can all this be simplified to:

```python
deps.[name]._add_dependency(deps[dname], dtypes)
```

That adds links in both directions.

Reply:
Basically.  I presumed the method wasn't called to avoid the check.